modules/web/src/common/components/objectmeta/component.ts

```diff
@@ -23,6 +26,8 @@ export class ObjectMetaComponent {
   @Input() initialized = false;
 
   private objectMeta_: ObjectMeta;
+  private readonly kdState_: KdStateService = GlobalServicesModule.injector.get(KdStateService);
```

Comment:
Why not using constructor for that?

Reply:
^ @marcosdiez

Reply:
@maciaszczykm I tried
```
  private readonly kdState_: KdStateService;

  constructor() {
    this.kdState_ = GlobalServicesModule.injector.get(KdStateService);
  }
```

and it fails just like without the constructor.

```
mdiez@batman:~/code/dashboard/modules/web$ yarn test src/resource/discovery/ingress/detail/component.spec.ts
 FAIL  src/resource/discovery/ingress/detail/component.spec.ts
  ● Test suite failed to run

    src/common/services/pluginloader/clientloader.service.ts:21:25 - error TS2339: Property 'System' does not exist on type 'Window & typeof globalThis'.

    21 const systemJS = window.System;
                               ~~~~~~
    src/common/services/pluginloader/clientloader.service.ts:31:14 - error TS2339: Property 'define' does not exist on type 'Window & typeof globalThis'.

    31       window.define(externalKey, [], () => {
                    ~~~~~~
    src/common/services/pluginloader/clientloader.service.ts:51:52 - error TS7006: Parameter 'm' implicitly has an 'any' type.

    51       return systemJS.import(dependency.path).then(m => {
                                                          ~
    src/common/services/pluginloader/clientloader.service.ts:52:16 - error TS7015: Element implicitly has an 'any' type because index expression is not of type 'number'.

    52         window['define'](dep, [], () => m.default);
                      ~~~~~~~~
    src/common/services/pluginloader/clientloader.service.ts:57:48 - error TS7006: Parameter 'module' implicitly has an 'any' type.

    57       return systemJS.import(plugin.path).then(module => module.default.default);
                                                      ~~~~~~

Test Suites: 1 failed, 1 total
```

I am out of ideas. I'll try again next week.

Reply:
@marcosdiez I mean like here: 

https://github.com/kubernetes/dashboard/blob/master/modules/web/src/common/components/actionbar/detailactions/exec/component.ts#L26

Reply:
@maciaszczykm it did not work either :(

Reply:
@marcosdiez Sorry, I do not have time this and perhaps next month to work on the project. Once I will find some I will take a look on it.